src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -1653,5 +1653,129 @@ TEST(writer_reader, error_reader_decode_packet) {
+
+        CHECK(writer.valid());
+        CHECK(reader.valid());
+
+        // try to resize writer with an invalid value
+        CHECK(!writer.resize(encoder.max_block_length() * 2));
+
+        // ensure that the block size was not updated
+        fill_all_packets(0);
+
+        // write packets to dispatcher
+        for (size_t i = 0; i < NumSourcePackets; ++i) {
+            writer.write(source_packets[i]);
+        }
+
+        // deliver packets from dispatcher to reader
+        dispatcher.push_written();
+
+        // read packets
+        for (size_t i = 0; i < NumSourcePackets; ++i) {
+            packet::PacketPtr p = reader.read();
+            CHECK(p);
+            check_audio_packet(p, i);
+            check_restored(p, false);
+        }
+
+        CHECK(reader.alive());
+        CHECK(dispatcher.source_size() == 0);
+        CHECK(dispatcher.repair_size() == 0);
+    }
+}
+
+TEST(writer_reader, reader_oversized_block) {
+    for (size_t n_scheme = 0; n_scheme < Test_n_fec_schemes; ++n_scheme) {
+        config.scheme = Test_fec_schemes[n_scheme];
+
+        OFEncoder encoder(config, FECPayloadSize, allocator);
+        OFDecoder decoder(config, FECPayloadSize, buffer_pool, allocator);
+
+        CHECK(encoder.valid());
+        CHECK(decoder.valid());
+
+        CHECK(decoder.max_block_length() == encoder.max_block_length());
+        CHECK((NumSourcePackets + NumRepairPackets) < encoder.max_block_length());
+
+        packet::Queue queue;
+        PacketDispatcher dispatcher(NumSourcePackets, NumRepairPackets);
+
+        // We are going to violates source_block_length field of a FEC packet,
+        // but Reed-Solomon does not allows us to set this field more than 255.
+        // As a result LDPC composer is used.
+        Writer writer(config, FECPayloadSize, encoder, queue, ldpc_source_composer,
+                      ldpc_repair_composer, packet_pool, buffer_pool, allocator);
+
+        Reader reader(config, decoder, dispatcher.source_reader(),
+                      dispatcher.repair_reader(), rtp_parser, packet_pool, allocator);
+
+        CHECK(writer.valid());
+        CHECK(reader.valid());
+
+        fill_all_packets(0);
+
+        // encode packets and write to queue
+        for (size_t i = 0; i < NumSourcePackets; ++i) {
+            writer.write(source_packets[i]);
+        }
+
+        // write packets from queue to dispatcher
+        for (size_t i = 0; i < NumSourcePackets + NumRepairPackets; ++i) {
+            packet::PacketPtr p = queue.read();
+            CHECK(p);
+            if (i == 5) {
+                // violates: SBL + RBL > MAX_NES (for source packets)
+                p->fec()->source_block_length = encoder.max_block_length() * 2;
+                // reset block length to ensure that our packet won't be dropped
+                p->fec()->block_length = 0;
+            }
+            if (i == NumSourcePackets + 3) {
+                // violates: SBL + RBL > MAX_NES (for repair packets)
```

Comment:
`SBL + RBL <= MAX_NES`

also I think you need to remove " (for repair packets)"

Reply:
Done